storage/gcsemu/memstore.go

```diff
@@ -149,6 +149,10 @@ func (ms *memstore) Copy(srcBucket string, srcFile string, dstBucket string, dst
 
 func (ms *memstore) Delete(bucket string, filename string) error {
 	if filename == "" {
+		if _, ok := ms.buckets[bucket]; !ok {
+			return os.ErrNotExist
+		}
+
 		// Remove the bucket
 		delete(ms.buckets, bucket)
 	} else if b := ms.getBucket(bucket); b != nil {
```

Comment:
I believe there's another case here when we try to delete a file and the bucket doesn't exist, we are currently returning 200

Reply:
You are fixing both here?

Reply:
I updated with this fix as well